instrumentation/micrometer/micrometer-1.5/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/micrometer/v1_5/TimerTest.java

```diff
@@ -0,0 +1,138 @@
+                metric ->
+                    assertThat(metric)
+                        .hasDescription("This is a test timer")
+                        .hasUnit("milliseconds")
+                        .hasDoubleHistogram()
+                        .points()
+                        .satisfiesExactly(
+                            point ->
+                                assertThat(point)
+                                    .hasSum(42_000)
+                                    .hasCount(1)
+                                    .attributes()
+                                    .containsOnly(attributeEntry("tag", "value")))));
+    testing.clearData();
+
+    // when
+    Metrics.globalRegistry.remove(timer);
+    timer.record(12, TimeUnit.SECONDS);
+
+    // then
+    testing.waitAndAssertMetrics(
+        INSTRUMENTATION_NAME,
+        "testTimer",
+        metrics ->
+            metrics.allSatisfy(
+                metric ->
+                    assertThat(metric)
+                        .hasDoubleHistogram()
+                        .points()
+                        .noneSatisfy(point -> assertThat(point).hasSum(54_000).hasCount(2))));
+  }
+
+  @Test
+  void testMicrometerHistogram() {
+    // given
+    Timer timer =
+        Timer.builder("testHistogram")
+            .description("This is a test timer")
+            .tags("tag", "value")
+            .serviceLevelObjectives(
+                Duration.ofSeconds(1),
+                Duration.ofSeconds(10),
+                Duration.ofSeconds(100),
+                Duration.ofSeconds(1000))
+            .distributionStatisticBufferLength(10)
+            .register(Metrics.globalRegistry);
+
+    // when
+    timer.record(500, TimeUnit.MILLISECONDS);
+    timer.record(5, TimeUnit.SECONDS);
+    timer.record(50, TimeUnit.SECONDS);
+    timer.record(500, TimeUnit.SECONDS);
+
+    // then
+    testing.waitAndAssertMetrics(
+        INSTRUMENTATION_NAME,
+        "testHistogram.histogram",
+        metrics ->
+            metrics.anySatisfy(
+                metric ->
+                    assertThat(metric)
+                        .hasDoubleGauge()
+                        .points()
+                        .anySatisfy(
+                            point ->
+                                assertThat(point)
+                                    .hasValue(1)
+                                    .attributes()
+                                    .containsEntry("le", "1000"))
+                // TODO: I have absolutely no idea why otel drops these points below
+                /*
+                                        .anySatisfy(
+                                            point ->
+                                                assertThat(point)
+                                                    .hasValue(2)
+                                                    .attributes()
+                                                    .containsEntry("le", "10000"))
```

Comment:
It looks like OTel drops all values for a particular gauge after the first one is registered, even if other observations have completely different attributes. Is that correct behavior?

Reply:
Okay, this is actually this: https://github.com/open-telemetry/opentelemetry-java/issues/4005